### PR TITLE
Fix OL list items not increasing

### PR DESCRIPTION
**Pull request by JohnMcLear**
#4748 is a bug report where ordered list items were not properly formatted due to a lazy check.  The lazy check was ``x == 1`` and when we linted we changed it to ``x === 1``, the ``1`` was coming from a string check so it was typeOf string which meant when we did a more strict check it never returned true causing line number snot to incremented.

This PR.

* Fixes bug by parsing the string to an integer
* Fixes test by adding a new test which covers this edge case, although I'm not sure why it wasn't failing before on other tests..   we can investigate why once this is merged and we have order restored so can release.

### CODE REVIEW

File: src/static/js/domline.js [hunk 1]
```diff
@@ -101,7 +101,7 @@ domline.createDomLine = (nonEmpty, doesWrap, optBrowser, optDocument) => {
             postHtml = `</li></ul>${postHtml}`;
           } else {
             if (start) { // is it a start of a list with more than one item in?
-              if (start[1] === 1) { // if its the first one at this level?
+              if (parseInt(start[1]) === 1) { // if its the first one at this level?
```
Comment (rhansen) [inline src/static/js/domline.js:104]: Nit:

```suggestion
              if (Number.parseInt(start[1]) === 1) { // if its the first one at this level?
```

**Review by rhansen — APPROVED** [review]
